sdk/core/core/src/az_span.c

```diff
@@ -15,20 +17,42 @@ enum
   AZ_ASCII_LOWER_DIF = 'a' - 'A',
 };
 
-AZ_NODISCARD az_result
-az_span_slice(az_span span, int32_t low_index, int32_t high_index, az_span* out_sub_span)
+void az_precondition_failed_default()
 {
-  // left part
-  az_span left = span;
-  if (high_index >= 0)
+  /* By default, when a precondition fails the calling thread is suspended forever */
+  while (true)
   {
-    left = az_span_take(span, high_index);
+    az_platform_sleep_msec(INT32_MAX);
   }
-  az_span const right = az_span_drop(left, low_index);
-  int32_t new_length = az_span_length(right);
-  *out_sub_span = az_span_init(az_span_ptr(right), new_length, new_length);
+}
 
-  return AZ_OK;
+az_precondition_failed az_precondition_failed_callback = az_precondition_failed_default;
```

Comment:
Define this in the header.  We want customers to be able to replace the callback

Reply:
it is defined in header as extern
```c
extern az_precondition_failed az_precondition_failed_callback;
```
So anyone can change it's value at any time